src/Google/Cache/File.php

```diff
@@ -151,11 +151,14 @@ private function getCacheDir($file, $forWrite)
     // use the first 2 characters of the hash as a directory prefix
     // this should prevent slowdowns due to huge directory listings
     // and thus give some basic amount of scalability
-    $dirHash = substr(md5($file), 0, 2);
+    $fileHash = substr(md5($file), 0, 2);
+    $userHash = substr(md5(get_current_user()), 0, 2);
```

Comment:
Can you just use the whole username? The first 2 letters of the md5 hash of the usernames easily collision.